src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,176 @@
+                time.sleep(cooldown)
+
+            elif 500 <= status_code < 600:  # Error en el servidor
+                print(f"Error en el servidor ({status_code}). Reintentando...")
+                attempt += 1
+                time.sleep(cooldown)
+
+            else:
+                response.raise_for_status()
+
+        except requests.exceptions.RequestException as e:
+            print(f"Error al realizar la solicitud: {e}")
+            attempt += 1
+            time.sleep(cooldown)
+
+    raise Exception(f"La solicitud a la API falló después de {retries} intentos.")
+
+
+# -----------------------
+# Params: data = API schema
+# Return: data_proc = DataFrame[time,temperature_2m]
+# ----------------------
+def process_data(data: Dict[str, Any], magnitude: str) -> DataFrame:
+    time = data["hourly"]["time"]
+    temperature = data["hourly"][magnitude]
+
+    # DataFrame transformation
+    df = pd.DataFrame({"time": pd.to_datetime(time), magnitude: temperature})
+    df["time"] = pd.to_datetime(df["time"])
+    df.set_index("time", inplace=True)
+
+    return df
+
+
+def mean_calc(df: DataFrame, period: str) -> DataFrame:
+    mean_df = df.resample(period).mean()
+    return mean_df
+
+
+def sum_calc(df: DataFrame, period: str) -> DataFrame:
+    sum_df = df.resample(period).sum()
+    return sum_df
+
+
+def max_calc(df: DataFrame, period: str) -> DataFrame:
+    max_df = df.resample(period).max()
+    return max_df
+
+
+def plotting_line(df, magnitude, city):
+    color = PLOT_INFO[magnitude]["color"]
+    # Create a line plot using plotly express
+    fig = px.line(
+        df,
+        x=df.index,
+        y=magnitude,
+        markers=True,  # Adds markers to the line plot
+        title=f"{city}",
+        labels={
+            "x": "Date (Quarter)",
+            magnitude: f'{PLOT_INFO[magnitude]["y_label"]} ',
+        },
+    )
+
+    # Customize the layout for the plot
+    fig.update_traces(line_color=color)
+    fig.update_layout(
+        xaxis_title="Date (Quarter)",
+        yaxis_title=f'{PLOT_INFO[magnitude]["y_label"]}',
+        title_font_size=18,
+        xaxis_tickangle=-45,
+        autosize=True,
+        width=800,
+        height=500,
+    )
+
+    # fig.show()
+    return fig
+
+
+def main() -> None:
+    st.set_page_config(
+        layout="wide",
+    )
+
+    st.title("Meteorological Data Visualization")
+    cols = st.columns(3)
+    index = 0
+    for city in COORDINATES.keys():
+        print(f"Processing data for {city}")
+        # Get API data [time, temperature_2m]
+        data = get_data_meteo_api(city, VARIABLES[0])
+        # Process data
+        data_process = process_data(data, VARIABLES[0])
+        data_rep = mean_calc(data_process, "3ME")
+        # Plotting
+        with cols[index % 3]:  # Distribute plots in columns
+            fig = plotting_line(data_rep, VARIABLES[0], city)
+            st.plotly_chart(fig)
+
+        # Precipitation
```

Comment:
DRY again. It's basically the same code repeated once per variable. Think what if you need to extend it to 12 variables now...